src/main/java/frc/robot/subsystems/swerve/io/SwerveModuleIOSim.java

```diff
@@ -0,0 +1,82 @@
+package frc.robot.subsystems.swerve.io;
+
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.system.plant.DCMotor;
+import edu.wpi.first.wpilibj.simulation.FlywheelSim;
+import frc.robot.subsystems.swerve.Converstions;
+import frc.robot.subsystems.swerve.SwerveContants;
+import frc.robot.utils.fields.FieldsTable;
+
+public class SwerveModuleIOSim extends SwerveModuleIO {
+    private final FlywheelSim driveMotor;
+    private final FlywheelSim angleMotor;
+    private double encoderIntegratedDegreesSim = 0;
+    private double encoderAbsolueDegreesSim = 0;
+    private double distanceMeters = 0;
+    private final PIDController pidControllerAngle = new PIDController(1, 0, 0);
+
+    public SwerveModuleIOSim(FieldsTable fieldsTable, int driveMotorID, int angleMotorID, int encoderID) {
+        super(fieldsTable);
+
+        driveMotor = new FlywheelSim(DCMotor.getFalcon500(1), SwerveContants.GEAR_RATIO_DRIVE, 0.05);
+        angleMotor = new FlywheelSim(DCMotor.getFalcon500(1), SwerveContants.GEAR_RATIO_ANGLE, 0.004);
+    }
+
+    @Override
+    public void periodicBeforeFields() {
+        fields.recordOutput("angular velucity rad per sec", angleMotor.getAngularVelocityRadPerSec());
+        fields.recordOutput("MPS drive", driveMotor.getAngularVelocityRPM() * SwerveContants.WHEEL_CIRCUMFERENCE);
```

Comment:
This calculation is wrong, you get minutes and refer to it as it was seconds. Also, do not use the WHEEL_CIRCUMFERENCE but the WHEEL_RADIUS with radians.